Tasks/DeployAzureResourceGroup/AzureUtilityRest.ps1

```diff
@@ -0,0 +1,65 @@
+. "$PSScriptRoot/AzureUtilityGTE1.1.0.ps1"
+Import-Module $PSScriptRoot\ps_modules\VstsAzureRestHelpers_
+
+function Create-AzureResourceGroupIfNotExist
+{
+    param([string]$resourceGroupName,
+          [string]$location,
+          [object]$endpoint)
+
+    if(-not [string]::IsNullOrEmpty($resourceGroupName))
+    {
+        try
+        {
+            Write-Verbose "[Azure Resource Manager]Getting resource group:$resourceGroupName"
+            $azureResourceGroup = Get-AzRmResourceGroup $resourceGroupName $endpoint -ErrorAction silentlyContinue
+            Write-Verbose "[Azure Resource Manager]Got resource group:$resourceGroupName"
+        }
+        catch
+        {
+            #Ignoring the exception
+        }
+
+        if(-not $azureResourceGroup -and -not [string]::IsNullOrEmpty($location))
+        {
+            Write-Verbose "[Azure Resource Manager]Creating resource group $resourceGroupName in $location"
+            $azureResourceGroup = New-AzureRMResourceGroup -Name $resourceGroupName -Location $location -Verbose -ErrorAction Stop
+            Write-Host (Get-VstsLocString -Key "ARG_CreatedResourceGroup" -ArgumentList $resourceGroupName)
+        }
+        return $azureResourceGroup
+    }
+}
+
+function Get-AzureMachineCustomScriptExtension
```

Comment:
Is the response json of these calls matching with the azure ps cmdlet responses?